app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
I think it should be the same permission as show. If you can read this topic, then you can subscribe.